jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -187,76 +211,209 @@ MavenProject getProject() {
     return Preconditions.checkNotNull(project);
   }
 
+  /**
+   * Gets the base image reference.
+   *
+   * @return the configured base image reference
+   */
   String getBaseImage() {
+    if (System.getProperty(PropertyNames.fromImage) != null) {
+      return System.getProperty(PropertyNames.fromImage);
+    }
     return Preconditions.checkNotNull(Preconditions.checkNotNull(from).image);
   }
 
+  /**
+   * Gets the base image credential helper.
+   *
+   * @return the configured base image credential helper name
+   */
   @Nullable
   String getBaseImageCredentialHelperName() {
+    if (System.getProperty(PropertyNames.fromCredHelper) != null) {
+      return System.getProperty(PropertyNames.fromCredHelper);
+    }
     return Preconditions.checkNotNull(from).credHelper;
   }
 
   AuthConfiguration getBaseImageAuth() {
+    // System properties are handled in ConfigurationPropertyValidator
```

Comment:
We should probably move the property names passed to `ConfigurationPropertyValidator#getImageCredential` to `PropertyNames`